### PR TITLE
add jpush-react-native

### PR DESCRIPTION
**Pull request by huhuanming**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

**Comment by huhuanming** [comment]
@mhegazy  
it's pass all test.
Could you help me merge it?

**Comment by huhuanming** [comment]
A massive PR backlog need to do.
@mhegazy 
can I join in and help it?

**Comment by huhuanming** [comment]
@sandersn
sorry....

**Comment by huhuanming** [comment]
@DanielRosenwasser  😳 Could you help me merge it?

File: types/jpush-react-native/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    static clearNotificationById(id: string): void;
+
+    /**
+     * Android
+     */
+    static getInfo(cb: (map: any) => void): void;
+
+    static setTags(tags: string[], successCallback: (resultCode: number[]) => void, failedCallback: () => void): void;
+
+    static setAlias(alias: string, successCallback: (resultCode: number[]) => void, failedCallback: () => void): void;
```
Comment (DanielRosenwasser) [inline types/jpush-react-native/index.d.ts]: From the source, it looks like `successCallback` never gets called with parameters.

Comment (DanielRosenwasser) [inline types/jpush-react-native/index.d.ts]: What about `setAliasAndTags`?

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
Other than my comment this looks good and accurate. Let me know when you send out the changes and I can merge it in.

File: types/jpush-react-native/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+     * Android
+     */
+    static clearNotificationById(id: string): void;
+
+    /**
+     * Android
+     */
+    static getInfo(cb: (map: any) => void): void;
+
+    static setTags(tags: string[], successCallback: (resultCode: number[]) => void, failedCallback: () => void): void;
```
Comment (DanielRosenwasser) [inline types/jpush-react-native/index.d.ts]: From the source, it looks like `successCallback` never gets called with parameters.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by huhuanming** [comment]
Thank you for Reviewing

I fixed `successCallback` and  added `setAliasAndTags` .

**Comment by huhuanming** [comment]
@DanielRosenwasser 
😳 Can we go on?

**Pull request merged**